modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/SelectFileExtension.js

```diff
@@ -42,143 +41,18 @@ const nodeTreeArrayMapper = (nodeArray) => {
-
-	const [filterQuery, setFilterQuery] = useState('');
-
-	const selectedNodesRef = useRef(null);
-
-	const initialSelectedNodeIds = useMemo(() => {
-		const selectedNodes = [];
-
-		visit(nodes, (node) => {
-			if (node.selected) {
-				selectedNodes.push(node.id);
-			}
-		});
-
-		return selectedNodes;
-	}, [nodes]);
-
-	const handleSelectionChange = (selectedNodes) => {
-		const data = [];
-
-		// Mark newly selected nodes as selected.
-
-		visit(nodes, (node) => {
-			const isChildNode = !node.children;
-
-			if (selectedNodes.has(node.id) && isChildNode) {
-				data.push(node.id);
-			}
-		});
-
-		// Mark unselected nodes as unchecked.
-
-		if (selectedNodesRef.current) {
-			Object.entries(selectedNodesRef.current).forEach(([id, node]) => {
-				const nodeIndex = data.findIndex((node) => node.id === id);
-
-				if (!selectedNodes.has(id) && nodeIndex > -1) {
-					data[nodeIndex] = {
-						...node,
-						unchecked: true,
-					};
-				}
-			});
-		}
-
-		selectedNodesRef.current = data;
-
-		const openerWindow = Liferay.Util.getOpener();
-
-		openerWindow.Liferay.fire(itemSelectorSaveEvent, {data});
-	};
-
 	return (
-		<div className="select-file-extension">
-			<form
-				className="mb-4 pb-3 pt-3 select-file-extension-filter"
-				onSubmit={(event) => event.preventDefault()}
-				role="search"
-			>
-				<ClayLayout.ContainerFluid className="d-flex">
-					<div className="input-group">
-						<div className="input-group-item">
-							<input
-								className="form-control h-100 input-group-inset input-group-inset-after"
-								onChange={(event) =>
-									setFilterQuery(event.target.value)
-								}
-								placeholder={Liferay.Language.get('search')}
-								type="text"
-							/>
-
-							<div className="input-group-inset-item input-group-inset-item-after pr-3">
-								<ClayIcon symbol="search" />
-							</div>
-						</div>
-					</div>
-				</ClayLayout.ContainerFluid>
-			</form>
-
-			<form name={`${portletNamespace}selectSelectMimeTypeFm`}>
-				<ClayLayout.ContainerFluid containerElement="fieldset">
-					<div
-						className="type-tree"
-						id={`${portletNamespace}typeContainer`}
-					>
-						{nodes.length > 0 ? (
-							<Treeview
-								NodeComponent={Treeview.Card}
-								filter={getFilter(filterQuery)}
-								inheritSelection={true}
-								initialSelectedNodeIds={initialSelectedNodeIds}
-								multiSelection={true}
-								nodes={nodes}
-								onSelectedNodesChange={handleSelectionChange}
-							/>
-						) : (
-							<div className="border-0 pt-0 sheet taglib-empty-result-message">
-								<div className="taglib-empty-result-message-header"></div>
-								<div className="sheet-text text-center">
-									{Liferay.Language.get(
-										'no-file-extensions-were-found'
-									)}
-								</div>
-							</div>
-						)}
-					</div>
-				</ClayLayout.ContainerFluid>
-			</form>
-		</div>
+		<TreeFilter
+			itemSelectorSaveEvent={itemSelectorSaveEvent}
+			necessaryKeys={['id']}
```

Comment:
`necessaryKeys` I don't like very much this name 
Maybe can be use 'mandatory' instead on 'necessary' and add something about the purpose?
Something like `mandatoryBackendKeys`, `backendMandatoryProperties` ... I'm not sure

Reply:
Yeah, I struggled finding an appropriate name for that prop 🤣